socorro/external/es/super_search_fields.py

```diff
@@ -1069,7 +1001,6 @@ def apply_schema_properties(fields, schema):
         "namespace": "processed_crash",
         "query_type": "date",
         "storage_mapping": {
-            "format": "yyyy-MM-dd'T'HH:mm:ssZZ||yyyy-MM-dd'T'HH:mm:ss.SSSSSSZZ",
```

Comment:
adding this back causes test failures with errors like this:
```
elasticsearch.BadRequestError: BadRequestError(400, 'document_parsing_exception', "[1:267] failed to parse field [processed_crash.date_processed] of type [date] in document with id '50a15522-404d-4592-b536-5be222241106'. Preview of field's value: '2024-11-06T23:54:20+00:00'")
```

if I understand correctly, data validation outputting values like `'2024-11-06T23:54:20+00:00'` instead of the expected `'2024-11-06T23:54:20Z'`, which indicated to me that the old date format was no longer correct, and given tests pass without it I thought the default date format was now more correct

Reply:
mostly it's that i couldn't find an explicitly defined format that actually works. even setting this to `"yyyy-MM-dd'T'HH:mm:ss.SSSSSSZ"` still throws an error on `'2024-11-07T00:03:45.512909+00:00'`.

You can see the failure if you add back format and test analyzers:
```sh
make shell
ELASTICSEARCH_MODE=PREFER_NEW pytest -- socorro/tests/external/es/test_analyzers.py
```

Reply:
I was able to find two formats that pass tests: `"strict_date_optional_time||strict_date_optional_time_nanos"` or `"strict_date_optional_time_nanos"`. Those are optional time, but maybe that's fine? I can't get any custom value to work, even copying exact values from https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-date-format.html

Reply:
ah, I finally figured it out, it needed to have 5 `Z`s
changed to: `"yyyy-MM-dd'T'HH:mm:ssZZZZZ||yyyy-MM-dd'T'HH:mm:ss.SSSSSSZZZZZ"`